docs/ADR-4-Era-selectors-for-CLI-commands.md

```diff
@@ -0,0 +1,97 @@
+
+This would mean that submitting transactions built for an older era inherently carries with the risk that the transaction behaves other than intended.
+
+This is especially true for transactions built in an older era with an incomplete, made up or out-of-date values.
+
+This is also true for transactions upgraded from an older era, although to a lesser extent.
+
+Constructing a command in the currently era is the for any command that requires protocol-parametes is the only way to guarantee a transaction works as intended.
+
+Another thing that has changed since [[ADR-1 Default eras for CLI commands]] was adopted is the intruction of an era-based command structure.
+
+The era-based command structure looks like this: `cardano-cli selected-era command sub-command ...`.  For example `cardano-cli babbage transaction build ...`.
+
+We will for a time continue to support commands of the structure `cardano-cli command sub-command --selected-era` up until the Babbage era, but for Conway onwards, the era-based command structure must be used.
+
+The era-based command structure means that users when running a command must make some kind of decision upfront about what era they wish to run their command in.
+
+Such a structure also affords the possibility of having era selectors that follow an upgrade policy.  For example the `latest` era may select latest era that the CLI supports as a hard-coded value, defaulting to `babbage` whilst `conway` support is still in development.
+
+This ADR concerns what era selectors we will have and how they will behave.
+
+The decision should as much as possible take into account the following considerations:
+
+* Some users may care that CLI behaviour not suddenly change after a hard fork in a way that can break third party tools and user expectations.
+* Some users may care that CLI behaviour be such that transactions always behave as expected on-chain at the cost of breaking at a hard fork.
+* Historical precedent so as to avoid surprising SPOs and in a new release.
+* Simplicity and consistency.
+* Behaviour that just works where it makes sense.
+
+# Decision
+
+There will be no "default" era.  Instead, two era selectors will be included with the CLI:
+
+* `local` - This era is the latest hard-coded era supported by `cardano-cli` at the time `cardano-cli` was released.  This will not change in the event of a
+  hard-fork.  This command is offline and can run in the absence of a node socket.  For commands that require protocol parameters, this must be provided by
+  the user.  Any commands that are inherently online (eg `transaction submit`) will still use the same era as commands that are not inherently online.
+  In the event that the era selects an era different behind that of the network, `cardano-cli` will attempt to automically upgrade transaction to the network
```

Comment:
We have already decided we are not upgrading transactions so this part isn't relevant anymore.